### PR TITLE
hostpath snapshots use available minio image

### PR DESCRIPTION
**Pull request by laverya**
#### What type of PR is this?

type::feature

#### What this PR does / why we need it:
This PR unpins minio versions from kotsadm, allowing them to be updated independently - and allowing kURL to not ship a minio image with every kotsadm installation.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Kotsadm will use the existing minio image within the cluster to run hostpath snapshots, instead of pinning a specific version.
```

### CODE REVIEW

File: pkg/snapshot/filesystem.go [hunk 1]
```diff
@@ -276,17 +276,22 @@ func ensureFileSystemMinioDeployment(ctx context.Context, clientset kubernetes.I
 }
 
 func fileSystemMinioDeploymentResource(clientset kubernetes.Interface, secretChecksum string, deployOptions FileSystemDeployOptions, registryOptions kotsadmtypes.KotsadmOptions) (*appsv1.Deployment, error) {
```
Comment (jala-dx) [inline pkg/snapshot/filesystem.go:278]: Do we still want kots to deploy minio pods in the upcoming releases?

Comment (laverya) [reply pkg/snapshot/filesystem.go:278]: yes, to continue to support the people already using it

File: cmd/imagedeps/main_test.go [hunk 2]
```diff
@@ -37,16 +37,6 @@ func TestFunctional(t *testing.T) {
 		fn          tagFinderFn
 		expectError bool
 	}{
-		{
```
Comment (jala-dx) [inline cmd/imagedeps/main_test.go:40]: not sure why this code is removed.

Comment (laverya) [reply cmd/imagedeps/main_test.go:40]: because I was trying to remove minio entirely
I'm going to back out of doing that

**Review by jala-dx — APPROVED** [review]

**Comment by jala-dx** [comment]
merged with #2522

**Comment by laverya** [comment]
This is why you use normal merges 😆

**Pull request closed without merging**